src/Presentation/Nop.Web/Areas/Admin/Controllers/ProductController.cs

```diff
@@ -739,6 +739,48 @@ protected virtual async Task GenerateAttributeCombinationsAsync(Product product,
 
         #region Product list / create / edit / delete
 
+        public virtual async Task<IActionResult> ViewById(int id)
+        {
+            if (!await _permissionService.AuthorizeAsync(StandardPermissionProvider.ManageProducts))
+                return AccessDeniedView();
+
+            //try to get a product with the specified id
+            var product = await _productService.GetProductByIdAsync(id);
+            if (product == null || product.Deleted)
+                return RedirectToAction("List");
+
+            //a vendor should have access only to his products
+            var currentVendor = await _workContext.GetCurrentVendorAsync();
+            if (currentVendor != null && product.VendorId != currentVendor.Id)
+                return RedirectToAction("List");
+
+            //prepare model
+            var model = await _productModelFactory.PrepareProductModelAsync(null, product);
+
+            //try to get a product picture with the specified id
+            var productPicture = await _productService.GetProductPictureByIdAsync(model.Id)
+                                 ?? throw new ArgumentException("No product picture found with the specified id");
+
+            //fill in model values from the entity
+            var productPictureModel = productPicture.ToModel<ProductPictureModel>();
+
+            //fill in additional values (not existing in the entity)
+            var picture = (await _pictureService.GetPictureByIdAsync(productPicture.PictureId))
```

Comment:
same thing here, are you sure if product doens't have an image this will not throw an exception?